### ISSUE TITLE
I think GraphBacktracking searches too deep sometimes

### ISSUE THREAD

**Issue by wilfwilson** [peal/GraphBacktracking#6]
I am using my new refiner for stabiliser/transporter of a set of digraphs, which adds a single graph to the stack with extra vertices.

In a search for the stabiliser of a set of three graphs on 6 vertices, it is taking forever, and seems to be going to depth 20+ even though we're searching in S_6.

I think possibly the search is continuing beyond the bit when all of the points in `[1..6]` are already fixed, in particular, it must be branching on points outside of `[1..6]`.

The input looks something like this:
```
gap> o := Set(["&ECA@_OG", "&EQHcQHc", "&EHcQHcQ"], DigraphFromDigraph6String);
gap> GB_SimpleSearch(PartitionStack(6), [GB_Con.SetDigraphs(o, o)]);
```
(the refiner adds one big graph with 37 vertices in the initialise function, and that's it).

It's taking many minutes, here is the value of `base` at one interruption:
```
brk> rbase;
rec( branches := [ rec( cell := 3, pos := 25, tracer := <recording tracer of length 3> ),
      rec( cell := 4, pos := 26, tracer := <recording tracer of length 3> ),
      rec( cell := 1, pos := 1, tracer := <recording tracer of length 4> ),
      rec( cell := 6, pos := 2, tracer := <recording tracer of length 5> ),
      rec( cell := 7, pos := 3, tracer := <recording tracer of length 6> ),
      rec( cell := 8, pos := 4, tracer := <recording tracer of length 7> ),
      rec( cell := 9, pos := 5, tracer := <recording tracer of length 8> ),
      rec( cell := 2, pos := 7, tracer := <recording tracer of length 8> ),
      rec( cell := 11, pos := 8, tracer := <recording tracer of length 8> ),
      rec( cell := 12, pos := 9, tracer := <recording tracer of length 8> ),
      rec( cell := 13, pos := 10, tracer := <recording tracer of length 8> ),
      rec( cell := 14, pos := 11, tracer := <recording tracer of length 8> ),
      rec( cell := 15, pos := 12, tracer := <recording tracer of length 8> ),
      rec( cell := 16, pos := 13, tracer := <recording tracer of length 8> ),
      rec( cell := 17, pos := 14, tracer := <recording tracer of length 8> ),
      rec( cell := 18, pos := 15, tracer := <recording tracer of length 8> ),
      rec( cell := 19, pos := 16, tracer := <recording tracer of length 8> ),
      rec( cell := 20, pos := 17, tracer := <recording tracer of length 8> ),
      rec( cell := 21, pos := 18, tracer := <recording tracer of length 8> ),
      rec( cell := 22, pos := 19, tracer := <recording tracer of length 8> ),
      rec( cell := 23, pos := 20, tracer := <recording tracer of length 8> ),
      rec( cell := 24, pos := 21, tracer := <recording tracer of length 8> ),
      rec( cell := 25, pos := 22, tracer := <recording tracer of length 8> ),
      rec( cell := 26, pos := 23, tracer := <recording tracer of length 2> ) ], depth := 24,
  ps := [ [ 1 ], [ 7 ], [ 25 ], [ 26 ], [ 27 ], [ 2 ], [ 3 ], [ 4 ], [ 5 ], [ 6 ], [ 8 ], [ 9 ], [ 10 ], [ 11 \
], [ 12 ], [ 13 ], [ 14 ], [ 15 ], [ 16 ], [ 17 ], [ 18 ], [ 19 ], [ 20 ], [ 21 ], [ 22 ], [ 23 ], [ 24 ] ],
  root := rec( tracer := <recording tracer of length 6> ) )
brk>
```
And at other times I get things like:
```
brk> ps;
[ [ 1 ], [ 19 ], [ 25 ], [ 26 ], [ 27 ], [ 2 ], [ 3 ], [ 5 ], [ 4 ], [ 6 ], [ 14 ], [ 21 ], [ 23 ], [ 16 ], [ \
24 ], [ 13 ], [ 20 ], [ 15 ], [ 17 ], [ 22 ], [ 12 ], [ 7 ], [ 10 ], [ 8, 9, 11, 18 ] ]
```

**Comment by wilfwilson** [comment]
This can now be seen in the commented-out code in `tst/basic.tst`, which takes ages (haven't had termination yet):
```
gap> o := Set(["&ECA@_OG", "&EQHcQHc", "&EHcQHcQ"], DigraphFromDigraph6String);
gap> GB_SimpleSearch(PartitionStack(6), [GB_Con.SetDigraphs(o, o)]);
```

**Comment by wilfwilson** [comment]
Note that native GAP can sort this very quickly!
```
gap> Stabiliser(SymmetricGroup(6), o, OnSetsDigraphs);
Group([ (1,2,3,4,5,6), (1,5)(2,4)(3,6) ])
gap> time;
3
```

**Comment by wilfwilson** [comment]
@ChrisJefferson fixed this with a7fc450d4ad1999eeeb9ac26f0d1c4ce4d0bef57 and a corresponding change to BacktrackKit. Thanks!
